spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -191,6 +195,12 @@ private Broadcast<Map<String, DeleteFileSet>> broadcastRewritableDeletes() {
       return null;
     }
 
+    private boolean shouldBroadcastRewritableDeletes(TableOperations ops) {
```

Comment:
What about `shouldRewriteDeletes` as a shorter name? Also, I'd consider moving the logic for detecting whether DVs should be used to `SparkWriteConf`, where we can eventually call Eduard's utility method. We can then propagate this flag to the `Context` object within `SparkPositionDeltaWrite`.

```
private boolean shouldRewriteDeletes() {
  return context.useDVs() || context.deleteGranularity() == DeleteGranularity.FILE;
}
```

With a method in `SparkWriteConf`:

```
public boolean useDVs() {
  // call a utility method on Table
}
```

Reply:
The point I am trying to make is that accessing `TableOperations` here is too low-level.